docs/IMPLEMENTATION_DESIGN.md

```diff
@@ -1,114 +1,105 @@
 # Implementation design
 
-If you haven't done this before then please read the [features design](FEATURES_DESIGN.md) article first.
+If you haven't done this before, please read the [features design](FEATURES_DESIGN.md) article.
 
-It explains the "why"s of the 2 key design concepts of this app:
-* hierarchical configuration with inheritance, merging/overwriting and addivity,
+It explains the "why"s of the two key design concepts of this app:
+* hierarchical configuration with inheritance, merging/overwriting and additivity,
 * raw parameters passing
 
 Please also read the usual [contributing guide](CONTRIBUTION.md) for the basics of how to create PRs,
-the expected style of your code etc - the "general how"s.
+the expected style of your code etc. - the "general how"s.
 
 This article purpose is to explain "specific how"s - explain what is where in the code and how to add features
 by example.
 
-
 ## Structure
 
-Packages:
+### Packages:
   
-* `gitlabform.gitlabform` - contains the main app logic that provides the cli, parsing of the parameters,
-  uses below two packages (reads the config file, initializes connection to GitLab) to apply the config
+* `gitlabform.gitlabform` - contains the main app logic that provides the CLI, parsing of the parameters,
+  uses below two packages (reads the config file, initializes the connection to GitLab) to apply the config
   defined in `config.yml` to the requested set of projects.
 
 * `gitlabform.gitlab` - contains the GitLab API client code, grouped mostly by the GitLab API 
   [as documented at docs.gitlab.com](https://docs.gitlab.com/ee/api/api_resources.html).
 
 * `gitlabform.configuration` - contains the logic for reading the config from YAML file or a string
-  (the latter is used for testing only) and getting an effective config per GitLab project by merging
+  (the latter is only for testing) and getting an effective config per GitLab project by merging
   the config on these levels: common, group and project settings.
 
 ...and a trivial `bin/gitlabform` file that just runs the `gitlabform.gitlabform.GitLabForm.main()`.
 
-
 ## Common pattern - multiple inheritance
 
 In all of the above packages (except `gitlabform.gitlabform`, but it's an exception that will be refactored soon)
 we are using the _multiple inheritance_.
  
-The basic features are implemented in the "core" class defined in `core.py`. Extensions to it are defined in all other 
-files as "feature" classes that inherits the "core". Finally there is an "everything" class that groups them all - it is
-defined in `__init__.py` file.
-
+The basic features are implemented in the "core" class defined in `core.py`. Extensions to it are defined in all other files as "feature" classes that inherit the "core". Finally, there is an "everything" class that groups them all - it is
```

Comment:
Hey @vladdoster , thanks a lot for the PR! 😊

Just one remark: is there a reason you removed the line break here and in a few other places below?

Reply:
No 😞 . It was an oversight on my part. Fixing now.